src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -83,6 +86,122 @@ async function autofillDefaultTransactionValues(
   }
 }
 
+function createSimulatedTransaction(
+  blockchain: Blockchain,
+  options: EthereumInternalOptions,
+  common: Common,
+  transaction: Ethereum.Transaction,
+  simulationBlock: Block
+): SimulationTransaction {
+  const simulationBlockHeader = simulationBlock.header;
+
+  let gas: Quantity;
+  if (typeof transaction.gasLimit === "undefined") {
+    if (typeof transaction.gas !== "undefined") {
+      gas = Quantity.from(transaction.gas);
+    } else {
+      // eth_call isn't subject to regular transaction gas limits by default
+      gas = options.miner.callGasLimit;
+    }
+  } else {
+    gas = Quantity.from(transaction.gasLimit);
+  }
+
+  let data: Data;
+  if (typeof transaction.data === "undefined") {
+    if (typeof transaction.input !== "undefined") {
+      data = Data.from(transaction.input);
+    }
+  } else {
+    data = Data.from(transaction.data);
+  }
+
+  // eth_call doesn't validate that the transaction has a sufficient
+  // "effectiveGasPrice". however, if `maxPriorityFeePerGas` or
+  // `maxFeePerGas` values are set, the baseFeePerGas is used to calculate
+  // the effectiveGasPrice, which is used to calculate tx costs/refunds.
+  const baseFeePerGasBigInt =
+    simulationBlockHeader.baseFeePerGas !== undefined
+      ? simulationBlockHeader.baseFeePerGas.toBigInt()
+      : undefined;
+
+  let gasPrice: Quantity;
+  const hasGasPrice = typeof transaction.gasPrice !== "undefined";
+  // if the original block didn't have a `baseFeePerGas` (baseFeePerGasBigInt
+  // is undefined) then EIP-1559 was not active on that block and we can't use
+  // type 2 fee values (as they rely on the baseFee)
+  if (!common.isActivatedEIP(1559) || baseFeePerGasBigInt === undefined) {
+    gasPrice = hasGasPrice
+      ? Quantity.Zero
+      : Quantity.from(transaction.gasPrice);
+  } else {
+    const hasMaxFeePerGas = typeof transaction.maxFeePerGas !== "undefined";
+    const hasMaxPriorityFeePerGas =
+      typeof transaction.maxPriorityFeePerGas !== "undefined";
+
+    if (hasGasPrice && (hasMaxFeePerGas || hasMaxPriorityFeePerGas)) {
+      throw new Error(
+        "both gasPrice and (maxFeePerGas or maxPriorityFeePerGas) specified"
+      );
+    }
+    // User specified 1559 gas fields (or none), use those
+    let maxFeePerGas = 0n;
```

Comment:
nite: These could both be conditionals, which'll make it more concise.

e.g.
```suggestion
    let maxFeePerGas = hasMaxFeePerGas ? BigInt(transaction.maxFeePerGas) : 0n;
```

Reply:
Sure! Fixed.